src/SeleniumLibrary/keywords/javascript.py

```diff
@@ -73,19 +91,52 @@ def execute_async_javascript(self, *code):
         | ...         | window.setTimeout(answer, 2000);                |
         | `Should Be Equal` | ${result} | text |
         """
-        js = self._get_javascript_to_execute(code)
-        self.info("Executing Asynchronous JavaScript:\n%s" % js)
-        return self.driver.execute_async_script(js)
-
-    def _get_javascript_to_execute(self, lines):
-        code = ''.join(lines)
-        path = code.replace('/', os.sep)
-        if os.path.isabs(path) and os.path.isfile(path):
-            code = self._read_javascript_from_file(path)
-        return code
+        js_code, js_args = self._get_javascript_to_execute(code)
+        self.info('Executing Asynchronous JavaScript:\n%s\nBy using argument(s):\n"%s"'
+                  % (js_code, ', '.join(js_args)))
+        return self.driver.execute_async_script(js_code, *js_args)
+
+    def _get_javascript_to_execute(self, code):
+        js_code, js_args = self._separate_code_and_args(code)
+        if not js_code:
+            raise ValueError('JavaScript code was not found from code argument.')
+        js_code = ''.join(js_code)
+        path = js_code.replace('/', os.sep)
+        if os.path.isfile(path):
+            js_code = self._read_javascript_from_file(path)
+        return js_code, js_args
+
+    def _separate_code_and_args(self, code):
+        if not code:
+            raise ValueError('There must be at least one argument defined.')
+        js_code, js_args = [], []
+        get_code, get_args = False, False
+        found_code, found_args = False, False
+        for line in code:
+            if line == 'JAVASCRIPT' and found_code:
+                message = 'JAVASCRIPT marker was found two times in the code.'
+                raise ValueError(message)
+            if line == 'JAVASCRIPT' and not found_code:
+                get_code, found_code = True, True
+                get_args = False
+                continue
+            if line == 'ARGUMENTS' and found_args:
+                message = 'ARGUMENTS marker was found two times in the code.'
+                raise ValueError(message)
+            if line == 'ARGUMENTS' and not found_args:
+                get_code = False
+                get_args, found_args = True, True
+                continue
+            if not get_code and not get_args:
+                get_code, found_code = True, True
+            if get_code:
+                js_code.append(line)
+            if get_args:
+                js_args.append(line)
+        return js_code, js_args
```

Comment:
This looks a bit complicated. Don't have time to look at more closely could it be improved. Will look again later.

Reply:
Yes, it's complicated. At least raising an error could be moved to a separate method. Need to think what else could be done.